rust/core-lib/src/plugins/rpc_types.rs

```diff
@@ -127,10 +127,35 @@ pub struct Span {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
+pub struct ScopeSpan {
+    pub start: usize,
+    pub end: usize,
```

Comment:
At least for syntax highlighting, every single character will be covered by a span, so it would save space to make this just a `length`. But on the other hand the JSON format right now has like 5x the characters per span it could have. So this is fine for now but if serialization volume is ever an issue the format should be changed to something like `[len,id],[len,id],[len,id],...`.

By my estimation this format pushes the total size for all of jquery to 6mb (including the stacks), which depending on serialization overhead may not be bad.